js/ui/components/codingGameService.js

```diff
@@ -0,0 +1,137 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+
+const CodingGameService = imports.gi.CodingGameService;
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const Lang = imports.lang;
+const Meta = imports.gi.Meta;
+
+const CodingChatboxTextService = new Lang.Class({
+    Name: 'CodingChatboxTextService',
+
+    _init: function() {
+        this._signalHandlers = {};
+        this._eventHandlers = {
+            'move-window': Lang.bind(this, this._waitForNextWindowMove),
+            'stop-moving-windows': Lang.bind(this, this._waitForNoWindowsToMove)
+        };
+    },
+
+    _startListeningForEvent: function(name) {
+        if (this._eventHandlers[name]) {
+            this._disconnectHandlersFor(name);
+            this._eventHandlers[name]();
+        }
+    },
+
+    enable: function() {
+        // Connect to the service
+        this._controller = new CodingGameService.AppIntegrationController();
+        this._controller.service_event_with_listener('move-window', Lang.bind(this, function() {
+            this._startListeningForEvent('move-window');
+        }), Lang.bind(this, function() {
+            this._disconnectHandlersFor('move-window');
+        }));
+        this._controller.service_event_with_listener('stop-moving-windows', Lang.bind(this, function() {
+            this._startListeningForEvent('stop-moving-windows');
+        }), Lang.bind(this, function() {
+            this._disconnectHandlersFor('stop-moving-windows');
+        }));
+    },
+
+    disable: function() {
+        Object.keys(this._signalHandlers, Lang.bind(this, function(key) {
+            this._disconnectHandlersFor(key);
+        }));
+    },
+
+    _popSignalHandlers: function(name) {
+        name = name.replace(/\-/g, '_');
+        let retval = this._signalHandlers[name];
+        if (!retval) {
```

Comment:
While at this, we want to make coding style uniform with the rest of the shell; so please remove curly braces for single-line statements.